test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs

```diff
@@ -90,6 +90,20 @@ public void FastFilterWithSingleEqualsClause()
         Assert.IsFalse(fastFilter.Evaluate(s => "Test2"));
     }
 
+    [TestMethod]
+    public void AvoidingStackOverFlow()
```

Comment:
It would be better to split this into 2 tests (one for each method you are testing) and go for a more "functional" name (e.g. `ValidForPropertiesHandlesBigFilteringExpressions`).